lib/fdk/runner.rb

```diff
@@ -3,123 +3,53 @@
-            #         endbracket=true
-            #     elsif ls == "" && endbracket
-            #         # TODO: this isn't very robust, probably needs to be better :/
-            #         STDERR.puts "OBJECT: #{obs}"
-            #         payload = JSON.parse(obs)
-            #         STDERR.puts "payload: #{payload.inspect}"
-            #         c = Context.new(payload)
-            #         STDERR.puts "context: " + c.inspect
-            #         STDERR.flush
-            #         body = payload['body']
-            #         if c.content_type == 'application/json'
-            #             body = JSON.parse(body)
-            #         end
-            #         # TODO: begin/rescue so we can respond with proper error response and code
-            #         s = FDK.single_event(func, c, body)
-            #         response = {
-            #             headers: {
-            #                 'Content-Type' => 'application/json'
-            #             },
-            #             'status_code' => 200,
-            #             body: s.to_json,
-            #         }
-            #         STDOUT.puts response.to_json
-            #         STDOUT.puts
-            #         STDOUT.flush
-            #         obs = ""
-            #         next
-            #     else 
-            #         endbracket = false
-            #     end
-            #     obs += line
-            # end
-        elsif format == "default"
-            # TODO: check if content type json, and if so, parse it before passing it in
-            body = STDIN.read
-            payload = {}
-            payload['call_id'] = ENV['FN_CALL_ID']
-            payload['content_type'] = ENV['FN_HEADER_Content_Type']
-            payload['protocol'] = {
-                'type' => 'http',
-                'request_url' => ENV['FN_REQUEST_URL']
-            }
-            # STDERR.puts "payload: #{payload}"
-            c = Context.new(payload)
-            if c.content_type == "application/json"
-                # STDERR.puts "parsing json"
-                body = JSON.parse(body)
-            end
-            puts FDK.single_event(func, c, body).to_json
-        else
-            raise "Format '#{format}' not supported in Ruby FDK."
+      parser.on_parse_complete = lambda do |payload|
+        context = Context.new(payload)
+        body = payload['body']
+        if context.content_type == 'application/json' && body != ''
+          body = Yajl::Parser.parse(body)
         end
-    end
+        se = FDK.single_event(func, context, body)
+        response = {
+          headers: {
+            'Content-Type' => 'application/json'
+          },
+          'status_code' => 200,
+          body: se.to_json
+        }
+        STDOUT.puts response.to_json
+        STDOUT.puts
+        STDOUT.flush
+      end
 
-    def self.single_event(func, c, i)
-        s = send func, c, i
-        return s
+      STDIN.each_line { |line| parser.parse_chunk(line) }
```

Comment:
Think this can just be` parser.parse(STDIN)`

Reply:
That won't work for multiple calls, because it doesn't pull the separate docs off STDIN and process them individually.

If you read each_line from STDIN and parse them as chunks it does.